src/main/scala/viper/gobra/frontend/info/Info.scala

```diff
@@ -248,48 +252,95 @@ object Info extends LazyLogging {
+              pkg.imports.collectFirst {
+                case importNode if importNode.importPath == importPath => importNode
+              }.getOrElse(Violation.violation(s"could not find import node for import path $importPath"))
+          }
+          createImportError(importTarget, errs, importNode)
+      }.toVector
+      Left(messages)
+    }
+
     logger.trace {
       val durationS = f"${(System.currentTimeMillis() - startTimeMs) / 1000f}%.1f"
       s"computing errors for ${pkg.info.id} done, took ${durationS}s"
     }
 
-    val sourceNames = sources.map(_.name)
-    // use `sources` instead of `context.inputs` for reporting such that the message is correctly attributed in case of imports
-    config.reporter report TypeCheckDebugMessage(sourceNames, () => pkg, () => getDebugInfo(pkg, info))
-    if (errors.isEmpty) {
-      config.reporter report TypeCheckSuccessMessage(sourceNames, config.taskName, () => info, () => pkg, () => getErasedGhostCode(pkg, info), () => getGoifiedGhostCode(pkg, info))
-      Right(info)
-    } else {
-      // remove duplicates as errors related to imported packages might occur multiple times
-      // consider this: each error in an imported package is converted to an error at the import node with
-      // message 'Package <pkg name> contains errors'. If the imported package contains 2 errors then only a single error
-      // should be reported at the import node instead of two.
-      // however, the duplicate removal should happen after translation so that the error position is correctly
-      // taken into account for the equality check.
-      val typeErrors = pkg.positions.translate(errors, TypeError).distinct
-      config.reporter report TypeCheckFailureMessage(sourceNames, pkg.packageClause.id.name, () => pkg, typeErrors)
-      Left(typeErrors)
-    }
+    messagesOrInfo.fold(
+      messages => {
+        // remove duplicates as errors related to imported packages might occur multiple times
+        // consider this: each error in an imported package is converted to an error at the import node with
+        // message 'Package <pkg name> contains errors'. If the imported package contains 2 errors then only a single error
+        // should be reported at the import node instead of two.
+        // however, the duplicate removal should happen after translation so that the error position is correctly
+        // taken into account for the equality check.
+        val typeErrors = pkg.positions.translate(messages, TypeError).distinct
+        config.reporter report TypeCheckFailureMessage(sourceNames, pkg.packageClause.id.name, () => pkg, typeErrors)
+        Left(typeErrors)
+      },
+      info => {
+        config.reporter report TypeCheckSuccessMessage(sourceNames, config.taskName, () => info, () => pkg, () => getErasedGhostCode(pkg, info), () => getGoifiedGhostCode(pkg, info))
+        Right(info)
```

Comment:
Isn't this comment outdated now, given that we stop at the first package where we find errors?

Reply:
I've updated this comment